features/steps.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+from contextlib import closing
+import datetime
+import os
+from journal import INSERT_ENTRY
+from journal import connect_db
+from journal import DB_SCHEMA
+from lettuce import world
+from lettuce import step
+from lettuce import before
+from lettuce import after
+
+TEST_DSN = 'dbname=test_learning_journal user=roberthaskell'
+
+
+@before.all
+def db():
+    """set up a database"""
+    settings = {'db': TEST_DSN}
+    init_db(settings)
+    world.settings = settings
+
+
+@before.all
+def app():
+    """cofigure an app, then set up a fake server"""
+    from journal import main
+    from webtest import TestApp
+    os.environ['DATABASE_URL'] = TEST_DSN
+    app = main()
+    world.app = TestApp(app)
+
+
+@before.each_feature
+def enter_data(step):
+    """populate the database"""
+    entry('Test Title', 'Test Text')
+    entry('Uneditted Title', 'Unedited Text')
+    entry('Markdown Test', '### Header3')
+    mkdn = """
+    ```python
+        def thing():
+        pass ```
+    """
+    entry('Color Testing', mkdn)
+
+
+@after.all
+def clear_entries(settings):
+    """clear the database"""
+    with closing(connect_db(world.settings)) as db:
+        db.cursor().execute("DELETE FROM entries")
+        db.commit()
+
+
+@after.all
+def cleanup(step):
+    clear_db(world.settings)
+
+
+def init_db(settings):
+    with closing(connect_db(settings)) as db:
+        db.cursor().execute(DB_SCHEMA)
+        db.commit()
+
+
+def clear_db(settings):
+    with closing(connect_db(settings)) as db:
+        db.cursor().execute("DROP TABLE entries")
+        db.commit()
+
+
+def run_query(db, query, params=(), get_results=True):
+    cursor = db.cursor()
+    cursor.execute(query, params)
+    db.commit()
+    results = None
+    if get_results:
+        results = cursor.fetchall()
+    return results
+
+
+def entry(title, text):
+    """provide a single entry in the database"""
+    settings = world.settings
+    now = datetime.datetime.utcnow()
+    expected = (title, text, now)
+    with closing(connect_db(settings)) as db:
+        run_query(db, INSERT_ENTRY, expected, False)
+        db.commit()
+    world.expected = expected
+
+
+# Steps
+# Shared steps:
+@step('an entry with the title "(.*)"')
+def get_entry(step, title):
+    # get entry date, text, title, assign it to world
+    response = world.app.get('/')
+    assert title in response.body
+
+
+# Detail Feature
+@step('I press the detail button')
+def press_button(step):
+    # press the detail button associated with the entry held in world
+    world.response = world.app.get('/details/1')
```

Comment:
You should not be hard-coding this url into the test.  You should find the button in the page and click it.